datalad/support/annexrepo.py

```diff
@@ -2761,9 +2761,14 @@ def is_available(self, file_, remote=None, key=False, batch=False):
                 return False
         else:
             annex_cmd = ["checkpresentkey"] + ([remote] if remote else [])
-            out = self._batched.get(
-                ':'.join(annex_cmd), annex_cmd,
-                path=self.path)(key_)
+            try:
+                out = self._batched.get(
+                    ':'.join(annex_cmd), annex_cmd,
+                    path=self.path)(key_)
+            except CommandError:
+                # git-annex will signal an unknown remote by exiting with
+                # return code 1.
```

Comment:
hm, then shouldn't return_code be checked here and if not `1` then just re-raise?
Is return code `1` used for anything else?

Reply:
I am not sure, whether it is used for anything else. In non-batched mode `1` is used if the remote does not exist, or if the content is not present in the remote. There might be another error situation that is mapped onto returning `1`.

Reply:
I think to be inline with what this function did before, we want:

```
except CommandError as e:
    CapturedException(e)  # logs the exception
    out = e.stdout
```

and then proceed (potentially leading to a ValueError with a message). Any potential issue I can see with this, seems to be part of how this method behaves and has nothing to do with this PR.
Or do you see a case where this solution goes wrong, @christian-monch ?

(I seem unable to make this a committable "suggestion" as I don't get the right anchor point for it)

Reply:
I updated the comment in c5419974d4a9c9c015f68145d03eacaec7ce46f3 to describe more precisely what was meant, i.e. git-annex might still exit with non-zero in batch mode instead of returning errors in JSON-output records

Reply:
Adressed in datalad-internal call